lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1085,64 +1085,89 @@ def test_source_is_empty(self):
-                'hls': None,
                 'youtube': None,
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': None,
+                'desktop_mp4': None,
+                'desktop_webm': None,
             },
-            ['https://mp4.com/dm.mp4']
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
         ),
+    )
+    @ddt.unpack
+    @patch('xmodule.video_module.video_module.HLSPlaybackEnabledFlag.feature_enabled', Mock(return_value=True))
+    def test_val_encoding_in_context(self, val_video_encodings, video_url):
+        """
+        Tests that the val encodings correctly override the video url when the edx video id is set and
+        one or more encodings are present.
+        Accepted order of source priority is:
+            VAL's youtube source > external youtube source > hls > mp4 > webm.
+
+        Note that `https://www.youtube.com/watch?v=3_yD_cEKoCk` is the default youtube source with which
+        a video component is initialized. Current implementation considers this youtube source as a valid
+        external youtube source.
+        """
+        with patch('xmodule.video_module.video_module.edxval_api.get_urls_for_profiles') as get_urls_for_profiles:
+            get_urls_for_profiles.return_value = val_video_encodings
+            self.initialize_module(
+                data='<video display_name="Video" download_video="true" edx_video_id="12345-67890">[]</video>'
+            )
+            context = self.item_descriptor.get_context()
+        self.assertEqual(context['transcripts_basic_tab_metadata']['video_url']['value'], video_url)
+
+    @ddt.data(
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': None,
                 'youtube': None,
-                'desktop_mp4': None
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': 'https://mp4.com/dm.mp4',
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
-            ['https://webm.com/dw.webm']
+            ['https://hls.com/hls.m3u8']
         ),
         (
             {
-                'desktop_webm': None,
-                'hls': None,
-                'youtube': None,
-                'desktop_mp4': None
+                'youtube': 'v0TFmdO4ZP0',
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': None,
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
-            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
+            ['https://www.youtube.com/watch?v=v0TFmdO4ZP0']
         ),
     )
     @ddt.unpack
     @patch('xmodule.video_module.video_module.HLSPlaybackEnabledFlag.feature_enabled', Mock(return_value=True))
-    def test_val_encoding_in_context(self, val_video_encodings, video_url):
+    def test_val_encoding_in_context_without_external_youtube_source(self, val_video_encodings, video_url):
```

Comment:
can you merge these tests via `ddt`?

Reply:
Video initialization is different in these test cases. I think we can not simply merge them via `ddt` unless we repeat initialization parameters in each `ddt` value.

Reply:
lets feed `youtube_id_1_0`  to video xml viaa `ddt` as well. 

for example:  if `youtube_id_1_0=None` , video component will fallback to its default value.

Reply:
`youtube_id_1_0=None` means we are not using its default value. Initializing video without this param means component will fall back to its default value